### ISSUE TITLE
Yanking commandline only yanks last character in zsh

### ISSUE THREAD

**Issue by svenXY** [tmux-plugins/tmux-yank#63]
Hi,

I was also having the problem with the RPROMPT and started playing around with different themes from oh-my-zsh and - when there is no RPROMPT - prefix-y only yanks the last character of the commandline, so `echo foo` only yanks `o`. prefix-Y (yanking PWD) works fine though.

I'm using zsh in vi-modeand tmux with

```
set -g @shell_mode 'vi'
```

enabled

**Comment by aleprovencio** [comment]
You have to place the cursor on the first char, also have you seen https://github.com/tmux-plugins/tmux-yank/issues/48 ?

**Comment by svenXY** [comment]
yes, placing the cursor on the first character makes it work, but should this not be done by thew "Escape" "0" keys in the respective method?

**Comment by aleprovencio** [comment]
You got it :+1:

**Comment by docwhat** [comment]
Can you check the current `master` with `tmux` 2.4?

I'm pretty sure I fixed this in commit e01284962106089d39cbbff116a11dbfae19713f

**Comment by svenXY** [comment]
Nope, it still does not work as expected.

**Comment by docwhat** [comment]
Yeah, I'm not sure what's going on.  It is sending the <kbd>Escape</kbd>+<kbd>0</kbd> to go to beginning of line, but it isn't being honored?

**Comment by qadzek** [comment]
I experience the same bug using Tmux 3.3a and Zsh in Vi mode. Only the characters under and to the right of the cursor get copied. The other key bindings work without issue. No big deal though, I just have to remember to go to the beginning of the line first.

**Comment by swoh816** [comment]
After adding `set -g @shell_mode 'vi'` in `~/.tmux.conf`, the key binding `prefix + y` works while in the insert mode of vi mode (in zsh) WITHOUT needing to move the cursor to the beginning of the line in normal mode. It is as described in [README](https://github.com/tmux-plugins/tmux-yank/?tab=readme-ov-file#vi-mode-support), although my tmux version is 3.5a, which is supposed to work without adding the line according to the README.

**Comment by qadzek** [comment]
On my system, using Tmux 3.5a, `prefix + y` in Normal mode still does not work as expected, even after adding `set -g @shell_mode 'vi'`. In Insert mode, it yanks without issues.

As a workaround, you can use `V + y` in Normal mode.

**Comment by svenXY** [comment]
old and most probably no progress to be expected - closing
